src/solvers/lowering/byte_operators.cpp

```diff
@@ -918,7 +919,7 @@ static exprt unpack_rec(
     {
       extractbits_exprt extractbits(
         typecast_exprt::conditional_cast(
-          src, bv_typet{numeric_cast_v<std::size_t>(last_bit)}),
+          src, bv_typet{numeric_cast_v<std::size_t>(total_bits)}),
```

Comment:
How about factoring this out of the loop completely? It's the same expression for each iteration of the loop, so only constructing it once would actually be beneficial.

Reply:
@tautschnig I can pull that change into the first refactoring commit, that would probably be an improvement yes.

Reply:
Just make sure you don't hoist it too high up so that you incidentally fix the issue by mere refactoring :-)